### DISCUSSION TITLE
Self host Remix app

### DISCUSSION THREAD

**Discussion by arundish** [remix-run/remix#6823] [General]
My customers do not want to host on a public cloud. They host all their apps internally on a Windows IIS server. I have build a number of React apps for then with create-react-app. They just copy the dist folder to a IIS folder and it works. Node is installed on the server and I use Express for all backend work.

How do I deploy a Remix app on a IIS website? A stack overflow answer advises to copy the whole folder to the server, run `npm start` and  reverse proxy localhost to localhost:3000. I cannot do this since the server is logged off and even if I do, I will have to keep running this on every server restart.

If this is not possible, I will surely not be able to go ahead with Remix. I like the framework though.

**Comment by lili21** [comment]
you can use pm2 to handle the node process

**Comment by domus71** [comment]
You have to create a service to keep alive the process while you are logged off. You can use NSSM 2.24 to create your service.

 ```nssm.exe install <servicename>```

After, you can edit the service using node.exe and any argument you need (ex. index.js). It will be wise to use full path of nodejs exe and startup directory (the root directory of your app).

```nssm.exe edit <servicename>```
